gap/attributes/isomorph.gi

```diff
@@ -171,50 +190,150 @@ end);
 InstallMethod(IsomorphismSemigroups, "for finite monogenic semigroups",
 [IsMonogenicSemigroup and IsFinite, IsMonogenicSemigroup and IsFinite],
 function(S, T)
-  local genS, genT, SS, TT;
+  local s, t, SS, TT, map, inv;
+
+  if IsSimpleSemigroup(S) then
+    if IsSimpleSemigroup(T) then
+      return IsomorphismSemigroups(S, T);
+    else
+      return fail;
+    fi;
+  elif Size(S) <> Size(T) or NrDClasses(S) <> NrDClasses(T) then
+    return fail;
+  fi;
 
   # Monogenic semigroups are of the same size, are not groups, and have the
   # same number of D-classes by this point, and so they are isomorphism
-  genS := Representative(MaximalDClasses(S)[1]);
-  genT := Representative(MaximalDClasses(T)[1]);
-  SS := Semigroup(genS);
-  TT := Semigroup(genT);
-  return MagmaIsomorphismByFunctionsNC(S, T,
-           x -> genT ^ Length(Factorization(SS, x)),
-           x -> genS ^ Length(Factorization(TT, x)));
+  s := Representative(MaximalDClasses(S)[1]);
+  t := Representative(MaximalDClasses(T)[1]);
+  SS := Semigroup(s);
+  TT := Semigroup(t);
+  map := x -> t ^ Length(Factorization(SS, x));
+  inv := x -> s ^ Length(Factorization(TT, x));
+  return MagmaIsomorphismByFunctionsNC(S, T, map, inv);
 end);
 
+SEMIGROUPS.DigraphFromMultTable := function(S)
```

Comment:
Function name isn't fool-proofed, as proved by a fool.

Reply:
Two of us at least, renamed to "IsomorphismInvariantDigraph"